airflow/api_fastapi/execution_api/datamodels/asset.py

```diff
@@ -36,7 +41,48 @@ class AssetAliasResponse(BaseModel):
     group: str
 
 
-class AssetProfile(StrictBaseModel):
+class DagRunAssetReference(StrictBaseModel):
+    """DAGRun serializer for asset responses."""
+
+    run_id: str
+    dag_id: str
+    logical_date: datetime | None
+    start_date: datetime
+    end_date: datetime | None
+    state: str
+    data_interval_start: datetime | None
+    data_interval_end: datetime | None
+
+
+class AssetEventResponse(BaseModel):
+    """Asset event schema with fields that are needed for Runtime."""
+
+    id: int
+    asset_id: int
+    uri: str | None = Field(alias="uri", default=None)
+    name: str | None = Field(alias="name", default=None)
+    group: str | None = Field(alias="group", default=None)
+    extra: dict | None = None
+    source_task_id: str | None = None
+    source_dag_id: str | None = None
+    source_run_id: str | None = None
+    source_map_index: int
+    created_dagruns: list[DagRunAssetReference]
+    timestamp: datetime
+
+    @field_validator("extra", mode="after")
+    @classmethod
+    def redact_extra(cls, v: dict):
+        return redact(v)
+
+
+class AssetEventCollectionResponse(BaseModel):
+    """Collection of AssetEventResponse."""
+
+    asset_events: list[AssetEventResponse]
```

Comment:
Lets create a seperate asset_event.py datamodel for asset events? It will clutter this one

Reply:
sounds good!

Reply:
updated, thanks!